src/parser/parse_tree_cl.c

```diff
@@ -11362,7 +11362,7 @@ pt_print_expr (PARSER_CONTEXT * parser, PT_NODE * p)
       break;
     case PT_CAST:
       r1 = pt_print_bytes (parser, p->info.expr.arg1);
-      if (PT_EXPR_INFO_IS_FLAGED (p, PT_EXPR_INFO_CAST_COLL_MODIFIER))
+      if (p->info.expr.arg1 != NULL && PT_EXPR_INFO_IS_FLAGED (p, PT_EXPR_INFO_CAST_COLL_MODIFIER))
```

Comment:
Do you think it is really needed?

Reply:
Yes, otherwise we have the crash. Thus, arg1 is set to null during binding (attribute is not found in class). So, we need to protect arg1, also cast_type.

Reply:
I'd like to propose the change as follows:
```c
diff --git a/src/parser/parse_tree_cl.c b/src/parser/parse_tree_cl.c
index f2439a97b..0254006de 100644
--- a/src/parser/parse_tree_cl.c
+++ b/src/parser/parse_tree_cl.c
@@ -11370,7 +11370,11 @@ pt_print_expr (PARSER_CONTEXT * parser, PT_NODE * p)
 
          sprintf (buf, " collate %s", lang_get_collation_name (PT_GET_COLLATION_MODIFIER (p)));
 
-         if (p->info.expr.arg1->node_type == PT_VALUE)
+         if (p->info.expr.arg1 == NULL)
+           {
+             // do nothing for badly formed
+           }
+         else if (p->info.expr.arg1->node_type == PT_VALUE)
            {
              PT_NODE *v = p->info.expr.arg1;
              int v_coll_id;
```